main.go

```diff
@@ -1,49 +1,11 @@
 package main
 
 import (
-	"flag"
-	"fmt"
-	"os"
-	"sort"
-
-	"golang.org/x/tools/go/packages"
-
 	"github.com/polyfloyd/go-errorlint/errorlint"
+
+	"golang.org/x/tools/go/analysis/singlechecker"
 )
 
 func main() {
-	checkErrorf := flag.Bool("errorf", false, "Check whether fmt.Errorf uses the %w verb for formatting errors. See the readme for caveats")
-	flag.Parse()
-
-	cfg := &packages.Config{
-		Mode: packages.NeedTypes | packages.NeedTypesInfo,
-	}
-	pkgs, err := packages.Load(cfg, flag.Args()...)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "load: %v\n", err)
-		os.Exit(100)
-	}
-	if packages.PrintErrors(pkgs) > 0 {
-		os.Exit(100)
-	}
-
-	lints := []errorlint.Lint{}
-	for _, pkg := range pkgs {
-		if *checkErrorf {
-			l := errorlint.LintFmtErrorfCalls(pkg.Fset, *pkg.TypesInfo)
-			lints = append(lints, l...)
-		}
-		l := errorlint.LintErrorComparisons(pkg.Fset, *pkg.TypesInfo)
-		lints = append(lints, l...)
-		l = errorlint.LintErrorTypeAssertions(pkg.Fset, *pkg.TypesInfo)
-		lints = append(lints, l...)
-	}
-	sort.Sort(errorlint.ByPosition(lints))
-
-	for _, lint := range lints {
-		fmt.Fprintf(os.Stderr, "%s: %s\n", lint.Pos, lint.Message)
-	}
-	if len(lints) > 0 {
-		os.Exit(1)
-	}
+	singlechecker.Main(errorlint.NewAnalyzer())
```

Comment:
Rewriting main causes the following things to break:
* No exit status returned if lints are triggered
* The `-errorf` flag is missing

While I agree with your suggestion that main.go should be moved to cmd/*, this will also cause breakage as well :/

From Github's insights I can see people are using this tool in their CI pipelines. Could you please rewrite main to ensure these two behaviours are retained?

Reply:
Those two things does not break. 
`singlechecker` [exits with code](https://github.com/golang/tools/blob/ecd9fd270d5d8717504e3c25d668fe010450ea5e/go/analysis/singlechecker/singlechecker.go#L76) from internal [`checker.Run`](https://pkg.go.dev/golang.org/x/tools/go/analysis/internal/checker#Run):

> Run loads the packages specified by args using go/packages, then applies the specified analyzers to them. Analysis flags must already have been set. It provides most of the logic for the main functions of both the singlechecker and the multi-analysis commands. **It returns the appropriate exit code.**

`-errorf` flag moved to `Analyzer` declaration:
https://github.com/polyfloyd/go-errorlint/blob/6aadd0d689e8b519c427695ee6b3b5082ba95682/errorlint/analysis.go#L18-L25

So both things work!
Try it: 
```console
$ go build -o go-errorlint .
$ ./go-errorlint ./errorlint/testdata/src/fmterrorf 
$ echo $?
0
$ ./go-errorlint -errorf ./errorlint/testdata/src/fmterrorf 
/go-errorlint/errorlint/testdata/src/fmterrorf/fmterrorf.go:15:33: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/go-errorlint/errorlint/testdata/src/fmterrorf/fmterrorf.go:20:29: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/go-errorlint/errorlint/testdata/src/fmterrorf/fmterrorf.go:20:34: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/go-errorlint/errorlint/testdata/src/fmterrorf/fmterrorf.go:25:29: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/go-errorlint/errorlint/testdata/src/fmterrorf/fmterrorf.go:30:33: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
/go-errorlint/errorlint/testdata/src/fmterrorf/fmterrorf.go:35:33: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors
$ echo $?
3
```

Reply:
You're right! Not sure what went wrong when I tested this the first time...